### ISSUE TITLE
Use `pkg_build` to link dependencies

### ISSUE THREAD

**Issue by timothee-haudebourg** [Smithay/wayland-rs#266]
By default cargo only uses the `-lwayland-client` flag (for the client) and assume that `ld` will find it. It is not the case with the nix package manager, where the library cannot be found without using `pkg_build`.

One way to solve this problem is to use the [pkg_config](https://github.com/rust-lang/pkg-config-rs) crate in the `build.rs` script like this (for wayland-client):
```
extern crate pkg_config;
...
fn main() {
   ...
   pkg_config::Config::new().atleast_version("1").probe("wayland-client").unwrap();
}
```
I've tried it and that allowed me to use the client crate on nixos, however I don't know how to write a proper fix that will interact well with all the other crates and their features (in particular `dlopen`, etc.).

**Comment by vberger** [comment]
This should probably be done in `wayland-sys`, which contains all the FFI-related content.

I'm not sure how this would interact with dlopen either tbh...

**Comment by kchibisov** [comment]
I guess this was solved by #328
